rfc/mediachain-rfc-1.md

```diff
@@ -0,0 +1,435 @@
+to other artefacts.
+
+Each chain cell carries also carries one or more signatures by
+signatory entities in the system.
+
+Note that the types of cells that can appear in either type of
+chain it is not limited to the types of relationships described
+above. These are merely the minimum schema to support the
+functionality of the Mediachain.
+
+Keeping this in mind, we can model the chains with the following
+schema:
+```
+EntityChainCell =
+   <EntityUpdateCell>
+ | <EntityLinkCell>
+ | <Nil>
+
+EntityUpdateCell = {
+ "type" : "entityUpdate"
+ "chain" : <Reference>
+ "entity" : <Reference>
+ "signatures" : <Signatures>
+  <Key> : <Value> ... ; metadata updates
+ }
+
+EntityLinkCell = {
+ "type" : "entityLink"
+ "chain" : <Reference>
+ "entity" : <Reference>
+ "entityLink" : <Reference>
+ ["relationship" : <Reference>]
+ "signatures" : <Signatures>
+ <Key> : <Value> ... ; entity relationship metadata
+ }
+
+ArtefactChainCell =
+   <ArtefactUpdateCell>
+ | <ArtefactCreationCell>
+ | <ArtefactDerivationCell>
+ | <ArtefactOwnershipCell>
+ | <ArtefactReferenceCell>
+ | <Nil>
+ 
+ArtefactUpdateCell = {
+ "type" : "artefactUpdate"
+ "chain" : <Reference>
+ "artefact" : <Reference>
+ "signatures" : <Signatures>
+ <Key> : <Value> ... ; metadata updates
+ }
+
+ArtefactCreationCell = {
+ "type" : "artefactCreatedBy"
+ "chain" : <Reference>
+ "artefact" : <Reference>
+ "entity" : <Reference>
+ "signatures" : <Signatures>
+ <Key> : <Value> ... ; creation metadata
+ }
+
+ArtefactDerivationCell = {
+ "type" : "artefactDerivedBy"
+ "chain" : <Reference>
+ "artefact" : <Reference>
+ "artefactOrigin" : <Reference>
+ "signatures" : <Signatures>
+ <Key> : <Value> ... ; creation metadata
+ }
+
+ArtefactOwnershipCell = {
+ "type" : "artefactRightsOwnedBy"
+ "chain" : <Reference>
+ "artefact" : <Reference>
+ "entity" : <Reference>
+ "signatures" : <Signatures>
+ <Key> : <Value> ... ; IP ownership metadata
+ }
+
+ArtefactReferenceCell = {
+ "type" : "artefactReferencedBy"
+ "chain" : <Reference>
+ "artefact" : <Reference>
+ ["entity" : <Reference>]
+ ["url" : <URL>]
+ "signatures" : <Signatures>
+ <Key> : <Value> ... ; reference metadata
+  }
+```
+
+## Indexing and Querying the Mediachain
+
+### The Journal
+
+Mediachain data are persistent and available through the IPFS network.
+However they are not _discoverable_ without the help of an index which
+maintains references to entities, artefacts and chain heads.
+
+In order to make the mediachain indexable and discoverable we need to
+maintain a _Journal_ of updates to the data store. The journal is
+very similar to a blockchain in that it acts as a public ledger of all
+Mediachain data transactions in the system. By replaying the journal
+and fetching the data from IPFS, any node in the Internet can bootstrap
+an index that allows it to read and query the Mediachain.
+
+### Appending Data in the Mediachain
+
+The datastore initially is almost empty, containing only the `Nil` object
```

Comment:
yup. The first thing will put on IPFS will be Nil and then we'll have its hash for ever :)